src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -203,8 +204,21 @@ public IEnumerable<string> GetTestPlatformExtensions(IEnumerable<string> sources
         /// <inheritdoc/>
         public IEnumerable<string> GetTestSources(IEnumerable<string> sources)
         {
-            // We do not have scenario where full CLR tests are deployed to remote machine, so no need to udpate sources
-            return sources;
+            List<string> actualSources = new List<string>();
+
+            // We are doing this specifically for UWP, should we extract it out to some other utility?
```

Comment:
Absolutely. This code takes us back couple of years, albeit the `if` used to be `#if ` in those days.

Reply:
I'm thinking of moving this to UWP provider, & invoke it via reflection so they own always own the ownership of giving us the correct exe, but that would still mean Desktop Provider would need to understand what all packages types it can support, however it need to understand the logic to extract sources from that package. Thoughts?